packages/runtime-core/src/componentEmits.ts

```diff
@@ -177,7 +178,7 @@ export function emit(
       return
     }
     instance.emitted[handlerName] = true
-    callWithAsyncErrorHandling(
+    result = callWithAsyncErrorHandling(
```

Comment:
If `handler` and `onceHandler` are existing at the same time, the result will be overwritten.